docs/user-guide/deployments.md

```diff
@@ -0,0 +1,414 @@
+RollingUpdateStrategy:          1 max unavailable, 1 max surge, 0 min ready seconds
+OldReplicationControllers:      deploymentrc-1562004724 (3/3 replicas created)
+NewReplicationController:       <none>
+Events:
+  FirstSeen     LastSeen        Count   From                            SubobjectPath   Reason          Message
+  ─────────     ────────        ─────   ────                            ─────────────   ──────          ───────
+  10m           10m             1       {deployment-controller }                        ScalingRC       Scaled up rc deploymentrc-1975012602 to 3
+  2m            2m              1       {deployment-controller }                        ScalingRC       Scaled up rc deploymentrc-1562004724 to 1
+  2m            2m              1       {deployment-controller }                        ScalingRC       Scaled down rc deploymentrc-1975012602 to 1
+  1m            1m              1       {deployment-controller }                        ScalingRC       Scaled up rc deploymentrc-1562004724 to 3
+  1m            1m              1       {deployment-controller }                        ScalingRC       Scaled down rc deploymentrc-1975012602 to 0
+```
+
+Here we see that when we first created the deployment, it created an RC and scaled it up to 3 replicas directly.
+When we updated the deployment, it created a new RC and scaled it up to 1 and then scaled down the old RC by 1, so that at least 2 pods were available at all times.
+It then scaled up the new RC to 3 and when those pods were ready, it scaled down the old RC to 0.
+
+### Multiple Updates
+
+Each time a new deployment object is observed, a replication controller is
+created to bring up the desired pods if there is no existing RC doing so.
+Existing RCs controlling pods whose labels match `.spec.selector` but the
+template does not match `.spec.template` are scaled down.
+Eventually, the new RC will be scaled to `.spec.replicas` and all old RCs will
+be scaled to 0.
+If the user updates the deployment while an existing deployment was in progress,
+deployment will create a new RC as per the update and start scaling that up and
+will roll the RC that it was scaling up before in its list of old RCs and will
+start scaling it down.
+For example: If user creates a deployment to create 5 replicas of nginx:1.7.9.
+But then updates the deployment to create 5 replicas of nging:1.9.1, when only 3
+replicas of nginx:1.7.9 had been created, then deployment will immediately start
+killing the 3 nginx:1.7.9 pods that it had created and will start creating
+nginx:1.9.1 pods. It will not wait for 5 replicas of nginx:1.7.9 to be created
+before changing course.
+
+## Writing a Deployment Spec
+
+As with all other Kubernetes configs, a Deployment needs `apiVersion`, `kind`, and
+`metadata` fields.  For general information about working with config files,
+see [here](simple-yaml.md), [here](configuring-containers.md), and [here](working-with-resources.md).
```

Comment:
We're deleting simple-yaml.md.
https://github.com/kubernetes/kubernetes/issues/15355#issuecomment-150051506